src/test/java/BaseTest.java

```diff
@@ -89,6 +99,56 @@ public static void clickAvatarIcon() {
         avatarIcon.click();
 
     }
+    public void searchSong(String songTitle) throws InterruptedException {
+        WebElement searchField = driver.findElement(By.cssSelector("//input[@placeholder='Press F to search']"));
+        searchField.sendKeys(songTitle);
+        searchField.click();
+        Thread.sleep(2000);
+    }
+    public void grabASong() {
+        WebElement song = driver.findElement(By.xpath("article[title='Simon Mathewson by Music Insiders By Fma'] a[class='name']"));
+        WebElement playlist = driver.findElement(By.xpath("a[class='active']"));
+
+        Actions acts = new Actions(driver);
+        acts.clickAndHold(song)
+                .release(playlist)
+                .build()
+                .perform();
+
+        playlist.click();
+    }
+    public void viewAllSearchResults() throws InterruptedException{
+        WebElement viewAllSearchResults = driver.findElement(By.cssSelector("button[data-test='view-all-songs-btn"));
+        viewAllSearchResults.click();
+        Thread.sleep(2000);
+
+        clickAddToButton();
+        Thread.sleep(2000);
+    }
+    public static void clickAddToButton() throws InterruptedException {
+        WebElement addTo = driver.findElement(By.xpath("//button.btn-add-to"));
+        addTo.click();
+        Thread.sleep(2000);
+    }
+
+    public static void selectSong(String simonMathewson) throws InterruptedException {
+        WebElement selectSong = driver.findElement(By.cssSelector("section[id='songResultsWrapper'] tr:nth-child(2) td:nth-child(1)"));
+        selectSong.click();
+        Thread.sleep(2000);
+
+
+    }
+    public void selectPlaylistname(String playlistName) throws InterruptedException {
+        WebElement selectPlaylist = driver.findElement(By.xpath("//*[@id=\"playlists\"]/ul/li[5]"));
+
+        selectPlaylist.click();
+        Thread.sleep(2000);
```

Comment:
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.